src/libs/actions/User.ts

```diff
@@ -982,8 +986,32 @@ function clearUserErrorMessage() {
     Onyx.merge(ONYXKEYS.USER, {error: ''});
 }
 
-function setMuteAllSounds(isMutedAllSounds: boolean) {
-    Onyx.merge(ONYXKEYS.USER, {isMutedAllSounds});
+function togglePlatformMute(platform: Platform, mutedPlatforms: Partial<Record<Platform, true>>) {
+    const newMutedPlatforms = mutedPlatforms?.[platform]
+        ? {...mutedPlatforms, [platform]: undefined} // Remove platform if it's muted
+        : {...mutedPlatforms, [platform]: true}; // Add platform if it's not muted
+
+    const optimisticData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.SET,
```

Comment:
Is it better to use `MERGE` here without passing `mutedPlatforms` to this function `togglePlatformMute`?

Reply:
For NVPs, the server uses SET, so it's probably best to use SET here as well. That will keep things consistent.